### ISSUE TITLE
Generated .gitignore should always use forward slashes

### ISSUE THREAD

**Issue by gavinpc** [gittup/tup#235]
Even on Windows.  The `.gitignore` generated by Tup on Windows uses backslashes instead of forward slashes.  Following is the `.gitignore` from a little test project:

    ##### TUP GITIGNORE #####
    ##### Lines below automatically generated by Tup.
    ##### Do not edit.
    .tup
    \.gitignore
    \out

Notwithstanding the slash-related issues elsewhere in Tup for Windows (especially for cygwin users), this one is straightforward: git recognizes (and indeed specifies) forward slashes in the ignore file on all platforms.

Meanwhile, the above gives incorrect results.  For instance, if I have an `out` folder elsewhere in the project (besides the root), it will be ignored.

**Comment by naudhr** [comment]
A simple patch would be enough to address this issue:
```patch
diff --git a/src/tup/db.c b/src/tup/db.c
index 4ed1bd1..6074a83 100644
--- a/src/tup/db.c
+++ b/src/tup/db.c
@@ -2476,7 +2476,7 @@ static int get_generated_nodelist(char *dest, tupid_t dt, int *total_len)
 			rc = -1;
 			goto out_reset;
 		}
-		*p = PATH_SEP;
+		*p = '/'; // .gitignore format requires forward slashes
 		p++;
 		len = sqlite3_column_int(*stmt, 0);
 		memcpy(p, sqlite3_column_text(*stmt, 1), len)
```
However, the function stack called to generate a gitignore entry looks like
```
*p = PATH_SEP;
get_generated_nodelist()
tup_db_alloc_generated_nodelist()
gitignore()
```
And it is unclear from the names of the underlying functions that they are only used for gitignore tasks (though currently they are).
So, either the names should be amended in favour of smth like `get_generated_nodelist_for_gitignore()` or the separator symbol must be a parameter, in the case the `tup_db_alloc_generated_nodelist()` will some day be used for some other purposes.

**Comment by gittup** [comment]
I tried to fix this in 1d7d43b9b8f81869d68a6a1e15f0a31e10111088, and
cleaned up the db.c functions a bit. Let me know if there are still issues
with it.

Thanks!
-Mike
